cwrapper/CArachneWrapperCTest.c

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2017 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "CArachneWrapper.h"
+
+#include <errno.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+
+enum TestReturn { TEST_SKIP, TEST_PASS, TEST_FAIL };
+typedef enum TestReturn TestReturn;
+
+/**
+ * Functions below are used to test thread creation
+ */
+static void*
+funcCreateTest(void* arg) {
+    usleep(1000);
```

Comment:
We may want a comment here explaining why this sleep is useful for the test, since it delays the execution of the test by a full ms. :)

Reply:
What about: we added this usleep() because we wanted to verify the correctness of createThread() and join() by extending execution time?

Reply:
That sounds reasonable, but I think we can be more explicit. How about something closer to the following?

"We sleep here so that threads joining this thread must contend for the join lock with high probability."

This is not necessarily the best comment either, but it is more explicit about why we're doing the sleep.